probes/commitsInLast90Days/def.yml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: commitsInLast90Days
+short: Check whether the project has at least one commit per week over the last 90 days.
+motivation: >
+  A project which is not active might not be patched, have its dependencies patched, or be actively tested and used. However, a lack of active maintenance is not necessarily always a problem. Some software, especially smaller utility functions, does not normally need to be maintained. For example, a library that determines if an integer is even would not normally need maintenance unless an underlying implementation language definition changed. A lack of active maintenance should signal that potential users should investigate further to judge the situation. A project may not need further features or maintenance; In this case, the probe results can be disregarded.
+implementation: >
+  The implementation checks the number of commits made in the last 90 days by any user type.
+outcome:
+  - If the project has commits from the last 90 days, the probe returns one OutcomePositive with a "commitsWithinThreshold" value.
```

Comment:
nit: explain what "commitsWithinThreshold" means

Reply:
Done in https://github.com/ossf/scorecard/pull/3507/commits/e682e2c5e37a62658166bf028c7196699d95f401

Reply:
Is that accurate though? 
`commitsWithinThreshold` is the number of commits, not the look back period. 
Wouldn't this probe need:
- the 90 days removed from the name
- `lookBackDays` added
- the explanation for `commitsWithinThreshold` fixed?

Reply:
> Wouldn't this probe need:
>
> the 90 days removed from the name

FWIW I intentionally did not remove it, since previous discussions on the 90 days revolved around the other probe.

I could see an advantage of both including and excluding. If we include, it lowers the barrier of running the probe with a metric that is generally considered a good threshold. If we exclude, it might need a bit more changing to the purpose of the probe; We could call in `commitsInLastNDays`, but that requires the user to pass the number of days, which I believe is not supported ATM (please correct me if I am wrong).

I could also see the advantage of having both a 90-day defined probe and adding a granular one later. That way, users can pick, and one of the probes can deprecate if users don't need it.

Reply:
re: name: we could use something generic like `hasRecentCommits`? Then we add other variables inn `Value` and it gives us a chance to provide an option later for users to set the commits / lookback time.

Reply:
> FWIW I intentionally did not remove it, since previous discussions on the 90 days revolved around the other probe.

I had thought this was during one of our syncs (the pre-kubecon sync), but I could be misremembering.

> hasRecentCommits

sounds reasonable to me

Reply:
> * the 90 days removed from the name
> * `lookBackDays` added
> * the explanation for `commitsWithinThreshold` fixed?

All of this should be done with https://github.com/ossf/scorecard/pull/3507/commits/795d010bafad9d1246e5811ad4814fda93f17fa2.